calculator.js

```diff
@@ -1,5 +1,193 @@
-const exampleAdditionInput = {
+// MAIN calculate function 
+
+const calculate = function(input) {
+  if (typeof input.num1 !== "number" || typeof input.num2 !== "number" ) {
+    return 'First two inputs must be numbers.';
+  }
+  if (input.operation === 'add' || input.operation === '+') {
+    return add(input.num1, input.num2);
+
+  } else if (input.operation === 'subtract' || input.operation === "-") {
+    return sub(input.num1, input.num2);
+
+  } else if (input.operation === 'multiply' || input.operation === '*') {
+    return mult(input.num1, input.num2);
+
+  } else if (input.operation === 'divide' || input.operation === '/') {
+    return div(input.num1, input.num2);
+
+  } else if (input.operation === 'remainder' || input.operation === '%') {
+    return mod(input.num1, input.num2);
+
+  } else if (input.operation === 'exponate' || input.operation === '^') {
+    return exp(input.num1, input.num2);
+
+  } else {
+    if (typeof input.operation === "string") {
+      return `${input.operation} is not a valid operation.`;
+    } else {
+      return 'Please enter a valid operation.';
+    }
+    
+  }
+}
+
+// OPERATION functions
+
+const add = (num1, num2) => num1 + num2;
+
+const sub = (num1, num2) => num1 - num2;
+
+const mult = (num1, num2) => num1 * num2;
+
+const div = (num1, num2) => {
+  if (num2 === 0) {
+    return "Cannot divide by 0"
+  }
+  return num1 / num2;
+}
+
+const mod = (num1, num2) => num1 % num2;
+
+const exp = (num1, num2) => num1 ** num2;
+
+// TESTING with example inputs 
+
+// String operators:
```

Comment:
Nice work writing a comprehensive set of example input to test the functionality of your program.